pureMaterials/createPurematlib.py

```diff
@@ -6,73 +6,158 @@
 # -can be used for mixing homogenized regions
 # -generally impurities at <~1e-3 wt. percent (10 wppm) are removed from materials
 #   (except SS-316 steels may contain boron impurity)
-# 
+#
 #
 import os
+from typing import Dict, Union, Optional
 from pyne import material
 from pyne.material import Material, MultiMaterial
 from pyne.material_library import MaterialLibrary
+from pyne import nucname
 #
 #
 
-def make_mat(nucvec, density, citation, molecular_mass = None):
-    mat = Material(nucvec, density = density, metadata = {'citation' : citation})
+
+def enrich(
+    nucvec: Dict[int, float], old_key: int, new_values: Dict[int, float]
```

Comment:
If I understand the data structure properly (more info in the docstring would help), then `old_key` is always an element ID and not an isotope/nuclide ID - is that right?

1. let's choose a variable name that makes this clear - as well as better docstring
2. What happens if `nucvec` is defined in terms of nuclides/isotopes already?

Reply:
Yes old_key is an element string. If nucvec is defined in terms of nuclides/isotopes the enrichment function won't work and it'll make the materials the same as if the dictionary didn't include mass_enrichment. In a future PR I plan on including logging to alert the user when this happens.